.github/workflows/upload-assets-to-cdn.yml

```diff
@@ -0,0 +1,53 @@
+# Upload files in assets/product-icons & assets/legal to stage and prod S3 buckets for CDN distribution
+#
+# Requirements:
+# 1. Github OIDC Provider in destination AWS account - https://github.com/aws-actions/configure-aws-credentials
+# 2. IAM Role with write access to destination bucket paths, scoped to the repository running the action
+
+name: "CDN Asset Upload"
+on:
+  push:
+    branches:
+      - main
+    paths:
+      - 'assets/**/'
+      - '!assets/other/**'
+
+jobs:
+  Upload-assets-to-CDN:
+    runs-on: ubuntu-latest
+    steps:
+      - name: "Checkout repository"
+        uses: actions/checkout@v2
+
+      - name: "Configure Stage AWS credentials"
+        uses: aws-actions/configure-aws-credentials@master
+        with:
+          aws-region: us-east-1
+          role-to-assume: arn:aws:iam::142069644989:role/fxa-content-cdn-stage-asset-upload
+          role-session-name: CDNAssetUpload
+
+      - name: "Asset upload to stage CDN S3 bucket"
+        run: |
+          aws s3 cp --cache-control 'public,max-age=86400' --recursive assets/product-icons --exclude "*" --include "*.svg" --include "*.png" s3://fxa-content-cdn-stage-distbucket-bpquvfnty86g/product-icons
```

Comment:
To @hwine's notes, maybe this should be `sync` instead of `cp` (though we should very this won't do anything unexpected the first time we use this)

Reply:
I verified this works as intended on the stage bucket with a dryrun.

```
aws s3 sync --cache-control 'public,max-age=86400' --exclude "*" --include "*.pdf" --content-disposition attachment  assets/legal s3://fxa-content-cdn-stage-distbucket-bpquvfnty86g/legal --dryrun
(dryrun) upload: assets/legal/legal4.pdf to s3://fxa-content-cdn-stage-distbucket-bpquvfnty86g/legal/legal4.pdf
(dryrun) upload: assets/legal/test1.pdf to s3://fxa-content-cdn-stage-distbucket-bpquvfnty86g/legal/test1.pdf
(dryrun) upload: assets/legal/test2.pdf to s3://fxa-content-cdn-stage-distbucket-bpquvfnty86g/legal/test2.pdf
```

```
aws s3 sync --cache-control 'public,max-age=86400' --exclude "*" --include "*.svg" --include "*.png" assets/product-icons  s3://fxa-content-cdn-stage-distbucket-bpquvfnty86g/product-icons --dryrun
(dryrun) upload: assets/product-icons/test-icon1.svg to s3://fxa-content-cdn-stage-distbucket-bpquvfnty86g/product-icons/test-icon1.svg
(dryrun) upload: assets/product-icons/test-icon2.png to s3://fxa-content-cdn-stage-distbucket-bpquvfnty86g/product-icons/test-icon2.png
```